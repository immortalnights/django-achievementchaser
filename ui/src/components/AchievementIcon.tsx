import { Typography, Tooltip } from "@mui/material"
import dayjs from "dayjs"
import { getRelativeTime } from "../dayjsUtilities"
import BorderedImage from "./BorderedImage"

const AchievementTitle = ({
    achievement,
    unlockedDatetime,
}: {
    achievement: Achievement
    unlockedDatetime?: string
}) => {
    let subTitle: string | undefined
    if (unlockedDatetime) {
        const unlockedDate = dayjs(unlockedDatetime)
        subTitle = `Unlocked: ${unlockedDate.format(
            "MMM D, YYYY"
        )} (${getRelativeTime(unlockedDate)})`
    } else if (achievement.globalPercentage) {
        subTitle = `Difficulty: ${achievement.globalPercentage.toFixed(2)}%`
    }

    return (
        <>
            <Typography>{achievement.displayName}</Typography>
            {subTitle && <Typography fontSize="small">{subTitle}</Typography>}
        </>
    )
}

const AchievementIcon = ({
    achievement,
    unlockedDatetime,
}: {
    game: string
    achievement: Achievement
    unlockedDatetime?: string
}) => {
    const getAchievementIcon = (unlocked: boolean, achievement: Achievement) =>
        unlocked ? achievement.iconUrl : achievement.iconGrayUrl

    return (
        <Tooltip
            title={
                <AchievementTitle
                    achievement={achievement}
                    unlockedDatetime={unlockedDatetime}
                />
            }
            arrow
        >
            <div>
                <BorderedImage
                    src={getAchievementIcon(!!unlockedDatetime, achievement)}
                    style={{
                        width: 64,
                        height: 64,
                    }}
                />
            </div>
        </Tooltip>
    )
}

export default AchievementIcon
