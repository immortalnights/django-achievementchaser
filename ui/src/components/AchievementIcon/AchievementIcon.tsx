import { Typography, Tooltip } from "@mui/material"
import dayjs from "dayjs"
import { formatDate, getRelativeTime } from "@/dayjsUtilities"
import BorderedImage from "../BorderedImage"

const AchievementTitle = ({
    achievement,
    unlockedDatetime,
}: {
    achievement: Achievement
    unlockedDatetime?: string
}) => {
    let subTitle: string | undefined
    if (unlockedDatetime) {
        const unlockedDate = dayjs(unlockedDatetime)
        subTitle = `Unlocked: ${formatDate(unlockedDate)} (${getRelativeTime(
            unlockedDate
        )})`
    } else if (achievement.globalPercentage) {
        subTitle = `Difficulty: ${achievement.globalPercentage.toFixed(2)}%`
    }

    return (
        <>
            <Typography>{achievement.displayName}</Typography>
            {subTitle && <Typography fontSize="small">{subTitle}</Typography>}
        </>
    )
}

const AchievementIcon = ({
    achievement,
    unlocked,
}: {
    game: string
    achievement: Achievement
    unlocked?: boolean | string
}) => {
    const getAchievementIcon = (unlocked: boolean, achievement: Achievement) =>
        unlocked ? achievement.iconUrl : achievement.iconGrayUrl

    return (
        <Tooltip
            title={
                <AchievementTitle
                    achievement={achievement}
                    unlockedDatetime={
                        typeof unlocked === "string" ? unlocked : undefined
                    }
                />
            }
            arrow
        >
            <div>
                <BorderedImage
                    src={getAchievementIcon(!!unlocked, achievement)}
                    style={{
                        width: 64,
                        height: 64,
                    }}
                />
            </div>
        </Tooltip>
    )
}

export default AchievementIcon
