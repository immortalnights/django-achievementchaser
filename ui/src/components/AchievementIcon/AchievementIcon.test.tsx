import renderer from "react-test-renderer"
import { expect, test } from "vitest"
import AchievementIcon from "."
import { achievement } from "@test/data"

test("snapshot", () => {
    const achievementIcon = renderer.create(
        <AchievementIcon game="Test Game" achievement={achievement} />
    )
    expect(achievementIcon.toJSON()).toMatchSnapshot()

    const unlockedAchievementIcon = renderer.create(
        <AchievementIcon
            game="Test Game"
            achievement={achievement}
            unlocked="2015-03-09T01:17:41+00:00"
        />
    )
    expect(unlockedAchievementIcon.toJSON()).toMatchSnapshot()
})
