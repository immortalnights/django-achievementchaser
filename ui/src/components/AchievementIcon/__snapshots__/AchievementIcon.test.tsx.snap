// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`snapshot 1`] = `
[
  <div
    aria-label={null}
    aria-labelledby={null}
    className=""
    data-mui-internal-clone-element={true}
    onBlur={[Function]}
    onFocus={[Function]}
    onMouseLeave={[Function]}
    onMouseOver={[Function]}
    onTouchEnd={[Function]}
    onTouchStart={[Function]}
  >
    <style
      dangerouslySetInnerHTML={
        {
          "__html": ".css-mt3xud{box-shadow:0 8px 17px 2px rgba(0, 0, 0, 0.14),0 3px 14px 2px rgba(0, 0, 0, 0.12),0 5px 5px -3px rgba(0, 0, 0, 0.2);border:1px solid black;display:block;}",
        }
      }
      data-emotion="css mt3xud"
    />
    <img
      className="css-mt3xud"
      src="https://steamcdn-a.akamaihd.net/steamcommunity/public/images/apps/102600/da371af592225df340dee76602a585d282197bc7.jpg"
      style={
        {
          "height": 64,
          "width": 64,
        }
      }
    />
  </div>,
  <style
    dangerouslySetInnerHTML={
      {
        "__html": ".css-gq6qa3-MuiTooltip-popper{z-index:1500;pointer-events:none;pointer-events:auto;pointer-events:none;}.css-gq6qa3-MuiTooltip-popper[data-popper-placement*="bottom"] .MuiTooltip-arrow{top:0;margin-top:-0.71em;}.css-gq6qa3-MuiTooltip-popper[data-popper-placement*="bottom"] .MuiTooltip-arrow::before{transform-origin:0 100%;}.css-gq6qa3-MuiTooltip-popper[data-popper-placement*="top"] .MuiTooltip-arrow{bottom:0;margin-bottom:-0.71em;}.css-gq6qa3-MuiTooltip-popper[data-popper-placement*="top"] .MuiTooltip-arrow::before{transform-origin:100% 0;}.css-gq6qa3-MuiTooltip-popper[data-popper-placement*="right"] .MuiTooltip-arrow{height:1em;width:0.71em;}.css-gq6qa3-MuiTooltip-popper[data-popper-placement*="right"] .MuiTooltip-arrow::before{transform-origin:100% 100%;}.css-gq6qa3-MuiTooltip-popper[data-popper-placement*="left"] .MuiTooltip-arrow{height:1em;width:0.71em;}.css-gq6qa3-MuiTooltip-popper[data-popper-placement*="left"] .MuiTooltip-arrow::before{transform-origin:0 0;}.css-gq6qa3-MuiTooltip-popper[data-popper-placement*="right"] .MuiTooltip-arrow{left:0;margin-left:-0.71em;}.css-gq6qa3-MuiTooltip-popper[data-popper-placement*="left"] .MuiTooltip-arrow{right:0;margin-right:-0.71em;}",
      }
    }
    data-emotion="css gq6qa3-MuiTooltip-popper"
  />,
  <style
    dangerouslySetInnerHTML={
      {
        "__html": ".css-s8pgp9-MuiPopper-root-MuiTooltip-popper{z-index:1500;pointer-events:none;pointer-events:auto;pointer-events:none;}.css-s8pgp9-MuiPopper-root-MuiTooltip-popper[data-popper-placement*="bottom"] .MuiTooltip-arrow{top:0;margin-top:-0.71em;}.css-s8pgp9-MuiPopper-root-MuiTooltip-popper[data-popper-placement*="bottom"] .MuiTooltip-arrow::before{transform-origin:0 100%;}.css-s8pgp9-MuiPopper-root-MuiTooltip-popper[data-popper-placement*="top"] .MuiTooltip-arrow{bottom:0;margin-bottom:-0.71em;}.css-s8pgp9-MuiPopper-root-MuiTooltip-popper[data-popper-placement*="top"] .MuiTooltip-arrow::before{transform-origin:100% 0;}.css-s8pgp9-MuiPopper-root-MuiTooltip-popper[data-popper-placement*="right"] .MuiTooltip-arrow{height:1em;width:0.71em;}.css-s8pgp9-MuiPopper-root-MuiTooltip-popper[data-popper-placement*="right"] .MuiTooltip-arrow::before{transform-origin:100% 100%;}.css-s8pgp9-MuiPopper-root-MuiTooltip-popper[data-popper-placement*="left"] .MuiTooltip-arrow{height:1em;width:0.71em;}.css-s8pgp9-MuiPopper-root-MuiTooltip-popper[data-popper-placement*="left"] .MuiTooltip-arrow::before{transform-origin:0 0;}.css-s8pgp9-MuiPopper-root-MuiTooltip-popper[data-popper-placement*="right"] .MuiTooltip-arrow{left:0;margin-left:-0.71em;}.css-s8pgp9-MuiPopper-root-MuiTooltip-popper[data-popper-placement*="left"] .MuiTooltip-arrow{right:0;margin-right:-0.71em;}",
      }
    }
    data-emotion="css s8pgp9-MuiPopper-root-MuiTooltip-popper"
  />,
]
`;

exports[`snapshot 2`] = `
[
  <div
    aria-label={null}
    aria-labelledby={null}
    className=""
    data-mui-internal-clone-element={true}
    onBlur={[Function]}
    onFocus={[Function]}
    onMouseLeave={[Function]}
    onMouseOver={[Function]}
    onTouchEnd={[Function]}
    onTouchStart={[Function]}
  >
    <style
      dangerouslySetInnerHTML={
        {
          "__html": ".css-mt3xud{box-shadow:0 8px 17px 2px rgba(0, 0, 0, 0.14),0 3px 14px 2px rgba(0, 0, 0, 0.12),0 5px 5px -3px rgba(0, 0, 0, 0.2);border:1px solid black;display:block;}",
        }
      }
      data-emotion="css mt3xud"
    />
    <img
      className="css-mt3xud"
      src="https://steamcdn-a.akamaihd.net/steamcommunity/public/images/apps/102600/e6241e91c24d64c878fbcc339cd542e1c7216d0d.jpg"
      style={
        {
          "height": 64,
          "width": 64,
        }
      }
    />
  </div>,
  <style
    dangerouslySetInnerHTML={
      {
        "__html": ".css-gq6qa3-MuiTooltip-popper{z-index:1500;pointer-events:none;pointer-events:auto;pointer-events:none;}.css-gq6qa3-MuiTooltip-popper[data-popper-placement*="bottom"] .MuiTooltip-arrow{top:0;margin-top:-0.71em;}.css-gq6qa3-MuiTooltip-popper[data-popper-placement*="bottom"] .MuiTooltip-arrow::before{transform-origin:0 100%;}.css-gq6qa3-MuiTooltip-popper[data-popper-placement*="top"] .MuiTooltip-arrow{bottom:0;margin-bottom:-0.71em;}.css-gq6qa3-MuiTooltip-popper[data-popper-placement*="top"] .MuiTooltip-arrow::before{transform-origin:100% 0;}.css-gq6qa3-MuiTooltip-popper[data-popper-placement*="right"] .MuiTooltip-arrow{height:1em;width:0.71em;}.css-gq6qa3-MuiTooltip-popper[data-popper-placement*="right"] .MuiTooltip-arrow::before{transform-origin:100% 100%;}.css-gq6qa3-MuiTooltip-popper[data-popper-placement*="left"] .MuiTooltip-arrow{height:1em;width:0.71em;}.css-gq6qa3-MuiTooltip-popper[data-popper-placement*="left"] .MuiTooltip-arrow::before{transform-origin:0 0;}.css-gq6qa3-MuiTooltip-popper[data-popper-placement*="right"] .MuiTooltip-arrow{left:0;margin-left:-0.71em;}.css-gq6qa3-MuiTooltip-popper[data-popper-placement*="left"] .MuiTooltip-arrow{right:0;margin-right:-0.71em;}",
      }
    }
    data-emotion="css gq6qa3-MuiTooltip-popper"
  />,
  <style
    dangerouslySetInnerHTML={
      {
        "__html": ".css-s8pgp9-MuiPopper-root-MuiTooltip-popper{z-index:1500;pointer-events:none;pointer-events:auto;pointer-events:none;}.css-s8pgp9-MuiPopper-root-MuiTooltip-popper[data-popper-placement*="bottom"] .MuiTooltip-arrow{top:0;margin-top:-0.71em;}.css-s8pgp9-MuiPopper-root-MuiTooltip-popper[data-popper-placement*="bottom"] .MuiTooltip-arrow::before{transform-origin:0 100%;}.css-s8pgp9-MuiPopper-root-MuiTooltip-popper[data-popper-placement*="top"] .MuiTooltip-arrow{bottom:0;margin-bottom:-0.71em;}.css-s8pgp9-MuiPopper-root-MuiTooltip-popper[data-popper-placement*="top"] .MuiTooltip-arrow::before{transform-origin:100% 0;}.css-s8pgp9-MuiPopper-root-MuiTooltip-popper[data-popper-placement*="right"] .MuiTooltip-arrow{height:1em;width:0.71em;}.css-s8pgp9-MuiPopper-root-MuiTooltip-popper[data-popper-placement*="right"] .MuiTooltip-arrow::before{transform-origin:100% 100%;}.css-s8pgp9-MuiPopper-root-MuiTooltip-popper[data-popper-placement*="left"] .MuiTooltip-arrow{height:1em;width:0.71em;}.css-s8pgp9-MuiPopper-root-MuiTooltip-popper[data-popper-placement*="left"] .MuiTooltip-arrow::before{transform-origin:0 0;}.css-s8pgp9-MuiPopper-root-MuiTooltip-popper[data-popper-placement*="right"] .MuiTooltip-arrow{left:0;margin-left:-0.71em;}.css-s8pgp9-MuiPopper-root-MuiTooltip-popper[data-popper-placement*="left"] .MuiTooltip-arrow{right:0;margin-right:-0.71em;}",
      }
    }
    data-emotion="css s8pgp9-MuiPopper-root-MuiTooltip-popper"
  />,
]
`;
